js/post/update_post.js

```diff
@@ -428,4 +493,129 @@ function fillMisc() {
 
 }
 
+/**
+ * Updates Credit Count for each POSt.
+ **/
+function fillCreditCount() {
+    'use-strict'
+
+    specCount = creditCountSpec  + creditCount300and400['Spec'];
+    majCount = creditCountMaj + creditCount300and400['Maj'];
+    minCount = creditCountMin + (creditCount300and400['Min'] * 0.5);
+
+
+    fillSpecCreditCount(specCount);
+    fillMajCreditCount(majCount);
+    fillMinCreditCount(minCount);
+}
+
+/**
+ * Autofills the credit count for Specialist
+ * @param {number} specCount The credit count for Specialist
+**/
+function fillSpecCreditCount(specCount) {
+    if (specCount >= 12) {
+        $('#spec_creds').html('(12.0/12.0)');
+    } else {
+        $('#spec_creds').html('(' + (specCount).toFixed(1) + '/12.0)');
+    }
+}
+
+/**
+ * Autofills the credit count for Major
+ * @param {number} majCount The credit count for Major
+**/
+function fillMajCreditCount(majCount) {
+    if (majCount >= 8) {
+        $('#maj_creds').html('(8.0/8.0)')
+    } else {
+        $('#maj_creds').html('(' + (majCount).toFixed(1) + '/8.0)');
+    }
+}
+
+/**
+ * Autofills the credit count for Minor.
+ * @param {number} minCount The credit count for Minor
+**/
+function fillMinCreditCount(minCount) {
+    if (minCount >= 4) {
+        $('#min_creds').html('(4.0/4.0)')
+    } else {
+        $('#min_creds').html('(' + (minCount).toFixed(1) + '/4.0)');
+    }
+}
+
+/**
+ * Autofills extra 200-level courses for last Minor constraint.
+ * @param {number} index The textbox number we are at
+ * @param HTMLElement} min300s Array of textbox elements to fill
+**/
+function addExtraMinCourses(index, min300s) {
+    for (var m = 0; m < 3; m++) {
+        if(index === 3) {
```

Comment:
If you mean the `if(index === 3)`, that is a bound check to make sure we don't check for textboxes past the number of textboxes we have available to fill. The for loop assumes that none of the textboxes might be filled, so it loops from 0 to 2. But the index stops it early if some of the textboxes had been filled previously. 

(I should add more documentation for this).